### PR TITLE
fix(openai): Support report token usage in stream mode

### LINKED ISSUE #627: 🐛 Bug Report: Number of tokens not reported when streaming OpenAI/AOAI
### Which component is this bug for?

OpenAI Instrumentation

### 📜 Description

`llm.usage.prompt_tokens` and `llm.usage.total_tokens` are not reported when streaming is used. Per discussion in Slack, this may be because this information is not included in the response from OpenAI/Azure OpenAI. Especially when `TRACELOOP_TRACE_CONTENT` is `false`, this is an important statistic and would be required for derived fields like token throughput.

### 👟 Reproduction steps

The following code correctly reports the number of tokens:
```
prompt = "Say Hi!"

response_non_streaming = openai.ChatCompletion.create(
    model="gpt-3.5-turbo",
    messages=[{"role": "user", "content": prompt}]
)
```

The following code does not report the number of tokens:
```
prompt = "Say Hi!"

response_streaming = openai.ChatCompletion.create(
    model="gpt-3.5-turbo",
    messages=[{"role": "user", "content": prompt}],
    stream=True
)
```

### 👍 Expected behavior

`llm.usage.prompt_tokens` and `llm.usage.total_tokens` should be reported regardless of `stream` setting.

### 👎 Actual Behavior with Screenshots

Example with non-streamed ada:
image

Example with streaming gpt-3.5:
image

### 🤖 Python Version

_No response_

### 📃 Provide any additional context for the Bug.

If this value is not reported by OpenAI/Azure OpenAI directly, then it may be a design decision of whether it's worth counting in this library, or whether only values returned by the APIs should be reported.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by Humbertzhang**
Fixes #627 

- [x] I have added tests that cover my changes.
- [x] If adding a new instrumentation or changing an existing one, I've added screenshots from some observability platform showing the change.

* For Traces:
image

* For Metrics:
image

- [x] PR name follows conventional commits format: `feat(instrumentation): ...` or `fix(instrumentation): ...`.
- [x] (If applicable) I have updated the documentation accordingly.
Related doc PR: https://github.com/traceloop/docs/pull/24

### CODE REVIEW

**Comment by Humbertzhang** [comment]
CI error:
```
  E               vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/home/runner/work/openllmetry/openllmetry/packages/opentelemetry-instrumentation-openai/tests/traces/cassettes/test_chat/test_chat_streaming.yaml') in your current record mode ('none').
  E               No match for the request (<Request (GET) https://openaipublic.blob.core.windows.net/encodings/cl100k_base.tiktoken>) was found.
```

tiktoken need to download file at it first run, so vcr's none record mode maybe not suite for it...

**Comment by nirga** [comment]
@Humbertzhang you can tell VCR to ignore certain hostnames:
https://github.com/traceloop/openllmetry/blob/373f5c72269362422d86e4a1f4644661fe95401c/packages/opentelemetry-instrumentation-llamaindex/tests/conftest.py#L50

File: packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/__init__.py [hunk 1]
```diff
@@ -205,3 +229,29 @@ def model_as_dict(model):
         return model_as_dict(model.parse())
     else:
         return model
+
+
+def should_record_stream_token_usage():
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/__init__.py:234]: Let's put it as an optional parameter to the `__init__` of the instrumentation (I'd default it to `true`)? I think it's cleaner (and I want to move the `should_trace_content` to be like that as well)

Comment (Humbertzhang) [reply packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/__init__.py:234]: You mean move it to `Traceloop.init()`'s arg, right?

Comment (nirga) [reply packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/shared/__init__.py:234]: No, the instrumentation init. Then traceloop sdk can pass it yes

**Comment by Humbertzhang** [comment]
Hello @nirga, I have resolved the VCR issue by adding ignore_hosts. However, I'm not entirely sure what you mean by `"put it as an optional parameter to the __init__ of the instrumentation."`

Could you provide me an example? Alternatively, we could just merge this, and I can make modifications in a later PR after you've updated `should_trace_content`.

**Review by nirga — APPROVED** [review]

**Comment by nirga** [comment]
@Humbertzhang sure, thanks!

**Pull request merged**